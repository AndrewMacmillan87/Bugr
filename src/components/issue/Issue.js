import React, { useState } from "react";

import StatusBtn from "../btns/StatusBtn";
import StatusSelector from "../status_selector/StatusSelector";

import PropTypes from "prop-types";

import "./issue.css";

export const Issue = ({ id, description, completion, priority, stage }) => {
    const [showPriorities, setPriorities] = useState(false);
    const [showStages, setStages] = useState(false);
    const [showIssueOptions, setIssueOptions] = useState(false);
    const [showDateOptions, setDateOptions] = useState(false);

    const stageId = `stage-${id}`;
    const priorityId = `priority-${id}`;

    return (
        <div className='issue-container'>
            <div className='issue-description-container'>
                <div className='issue-description'>{description}</div>
                <i
                    className='issue-menu fas fa-ellipsis-v'
                    title='Options'
                    onClick={() => setIssueOptions(true)}></i>
                <div className='issue-options-temp-container'>
                    {showIssueOptions && (
                        <div className='issue-options-container'>
                            <i
                                className='remove-menu-options fas fa-times-circle'
                                onClick={() => setIssueOptions(false)}></i>
                            <ul className='issue-options-list'>
                                <li className='issue-option-list-element'>
                                    New issue
                                </li>
                                <li className='issue-option-list-element'>
                                    Edit
                                </li>
                                <li className='issue-option-list-element'>
                                    Delete
                                </li>
                                <li className='issue-option-list-element'>
                                    Notes
                                </li>
                            </ul>
                        </div>
                    )}
                </div>
            </div>
            <div className='issue-state-btn-container'>
                {showStages && (
                    <StatusSelector status='stage' onClick={setStages} />
                )}
                <StatusBtn
                    id={stageId}
                    value={stage}
                    onClick={() => {
                        setStages(true);
                        setPriorities(false);
                    }}
                />
                {showPriorities && (
                    <StatusSelector status='priority' onClick={setPriorities} />
                )}
                <StatusBtn
                    id={priorityId}
                    value={priority}
                    onClick={() => {
                        setPriorities(true);
                        setStages(false);
                    }}
                />
            </div>
            <div className='date-container'>
                <div className='completion-date'>{completion}</div>
                <i
                    className='issue-menu fas fa-ellipsis-v'
                    title='Date options'
                    onClick={() => setDateOptions(true)}></i>
                <div className='date-options-temp-container'>
                    {showDateOptions && (
                        <div className='date-options-container'>
                            <i
                                className='remove-menu-options fas fa-times-circle'
                                onClick={() => setDateOptions(false)}></i>
                            <ul className='date-options-list'>
                                <li className='date-option-list-element'>
                                    Edit
                                </li>
                                <li className='date-option-list-element'>
                                    Delete
                                </li>
                            </ul>
                        </div>
                    )}
                </div>
            </div>
        </div>
    );
};

Issue.propTypes = {
    id: PropTypes.string,
    description: PropTypes.string,
    completion: PropTypes.string,
    priority: PropTypes.string,
    stage: PropTypes.string,
};

export default Issue;
