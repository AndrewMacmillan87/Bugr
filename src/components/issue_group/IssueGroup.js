import React from "react";

import PropTypes from "prop-types";

import Issue from "../issue/Issue";
import "./issue_group.css";

export const IssueGroup = ({ title, issues }) => {
    return (
        <section className='issue-group-container'>
            <div className='issue-group-title-container'>
                <h1 className='issue-group-title'>{title}</h1>
                <i className='add-group fas fa-plus-circle'></i>
            </div>
            <div className='issue-box'>
                {issues &&
                    issues.map((issue) => {
                        return (
                            <Issue
                                key={issue.id}
                                description={issue.description}
                                completion={issue.finish_by}
                                priority={issue.priority}
                                stage={issue.stage}
                            />
                        );
                    })}
            </div>
        </section>
    );
};

IssueGroup.propTypes = {
    title: PropTypes.string,
    issues: PropTypes.array,
};

export default IssueGroup;
