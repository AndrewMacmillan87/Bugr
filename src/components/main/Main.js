import React, { useState, useEffect } from "react";

import IssueGroup from "../issue_group/IssueGroup";

import "./main.css";

const projectIssues = require("../../test_project_data/project_a.json");

const Main = () => {
    const [issueGroups, setIssueGroups] = useState([]);

    useEffect(() => {
        setIssueGroups(projectIssues["Project A"]);
    }, []);

    return (
        <main className='main-application-container'>
            {issueGroups &&
                issueGroups.map((issueGroup) => {
                    return (
                        <IssueGroup
                            key={issueGroup.id}
                            title={issueGroup.title}
                            issues={issueGroup.issues}
                        />
                    );
                })}
        </main>
    );
};

export default Main;
