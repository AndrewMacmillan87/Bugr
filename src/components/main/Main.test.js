import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Main from "./Main";
import IssueGroup from "../issue_group/IssueGroup";
import Issue from "../issue/Issue";

describe("Main component", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Main />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Renders correctly", () => {
        const tree = renderer.create(<Main />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("Main HTML elements", () => {
        it("should render a main element", () => {
            const wrapper = mount(<Main />);
            expect(wrapper.find("main").length).toEqual(1);
        });

        it("should render 2 IssueGroup component", () => {
            const wrapper = mount(<Main />);
            expect(wrapper.find(IssueGroup).length).toEqual(2);
        });

        it("should render 4 Issue component", () => {
            const wrapper = mount(<Main />);
            expect(wrapper.find(Issue).length).toEqual(4);
        });
    });
});
