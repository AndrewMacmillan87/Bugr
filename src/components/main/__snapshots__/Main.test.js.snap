// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Renders correctly 1`] = `
<main
  className="main-application-container"
/>
`;
